vendor: add snappy

BramGruneir opened the pull request:
This should fix the broken `vendor is missing some 3rd-party dependencies:` introduced with 56d6ed49dd83927cc1f195c40e47c27f7c2cef42

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/14448)
<!-- Reviewable:end -->

BramGruneir committed:
vendor: add snappy

tamird commented on glide.lock:
```diff
@@ -389,7 +391,7 @@ imports:
   - imports
 - name: google.golang.org/api
-  version: 16ab375f94503bfa0d19db78e96bffbe1a34354f
+  version: 48e49d1645e228d1c50c3d54fb476b2224477303
```
What is this commit? It doesn't seem to exist? https://github.com/google/google-api-go-client/commit/48e49d1645e228d1c50c3d54fb476b2224477303

dt replied:
```commit 48e49d1645e228d1c50c3d54fb476b2224477303 (HEAD -> refs/heads/master, refs/remotes/origin/master, refs/remotes/origin/HEAD)
Author: Victor Chudnovsky <vchudnov@google.com>
Date:   Mon Mar 27 10:36:32 2017 -0700

    all: regenerate APIs

    Change-Id: I8ef5e9ef05bb82cae26359d4b472bfd2e4160e8c
    Reviewed-on: https://code-review.googlesource.com/11690
    Reviewed-by: Victor Chudnovsky <vchudnov@google.com>```

BramGruneir commented:
Changes in glide.lock since HEAD~
Added:
 - github.com/golang/snappy
Changed:
 - cloud.google.com/go: daf945bb -> 343c80ab
   - https://github.com/GoogleCloudPlatform/google-cloud-go/compare/daf945bb...343c80ab
 - github.com/cenk/backoff: 3db60c81 -> 5d150e7e
   - https://github.com/cenk/backoff/compare/3db60c81...5d150e7e
 - github.com/google/go-github: d683a1c6 -> 3c8f26c1
   - https://github.com/google/go-github/compare/d683a1c6...3c8f26c1
 - github.com/mattn/go-isatty: 57fdcb98 -> fc9e8d8e
   - https://github.com/mattn/go-isatty/compare/57fdcb98...fc9e8d8e
 - github.com/rubyist/circuitbreaker: af958308 -> 9d2405c9
   - https://github.com/rubyist/circuitbreaker/compare/af958308...9d2405c9
 - github.com/satori/go.uuid: b061729a -> 5bf94b69
   - https://github.com/satori/go.uuid/compare/b061729a...5bf94b69
 - github.com/spf13/pflag: 9ff6c692 -> d16db1e5
   - https://github.com/spf13/pflag/compare/9ff6c692...d16db1e5
 - golang.org/x/text: f28f3672 -> fc7fa097
   - https://github.com/golang/text/compare/f28f3672...fc7fa097
 - golang.org/x/tools: 23ca8a26 -> d63e2b22
   - https://github.com/golang/tools/compare/23ca8a26...d63e2b22
 - google.golang.org/api: 16ab375f -> 48e49d16
   - https://github.com/google/google-api-go-client/compare/16ab375f...48e49d16
 - google.golang.org/appengine: b5c7c247 -> 56d253d1
   - https://github.com/golang/appengine/compare/b5c7c247...56d253d1

tamird commented on glide.lock:
```diff
@@ -302,13 +304,13 @@ imports:
   version: 39b969c322811a58be9ec8be9d65198d43d8ba82
 - name: github.com/rubyist/circuitbreaker
-  version: af9583088c3453556f6e4f580693ba8606ca459a
+  version: 9d2405c9a39876d9c85a8d6cc8ce16b63eb83610
```
FYI we should probably be using the new CallContext added here. https://github.com/rubyist/circuitbreaker/compare/af958308...9d2405c9

BramGruneir replied:
Feel free to pin this dep if you're going to use it.

tamird replied:
Huh? Usually, when diligently updating dependencies, I audit the changes to spot additions that we should obviously be taking advantage of, such as this one.

Perhaps you could do that as well, though in a new PR if you wish, since you're taking on this burden.

BramGruneir replied:
Which makes sense, but this PR was to unblock master builds by including snappy.  Feel free to file an issue about using the new function.

tamird commented:
Again, this is not a case of unsafety - they added a feature that we should
take advantage of.

If you want to change the meaning of floating deps to "upgrade without
auditing", I would prefer we not.

On Wed, Mar 29, 2017 at 2:15 PM, David Taylor <notifications@github.com>
wrote:

> *@dt* commented on this pull request.
> ------------------------------
>
> In glide.lock
> <https://github.com/cockroachdb/cockroach/pull/14448#discussion_r108748825>
> :
>
> > @@ -302,13 +304,13 @@ imports:
>  - name: github.com/robfig/glock
>    version: 39b969c322811a58be9ec8be9d65198d43d8ba82
>  - name: github.com/rubyist/circuitbreaker
> -  version: af9583088c3453556f6e4f580693ba8606ca459a
> +  version: 9d2405c9a39876d9c85a8d6cc8ce16b63eb83610
>
> Sounds like we need to pin it then, so it only changes when someone wants
> it to, and is prepared to diligently audit the changes, if it is unsafe to
> float during other changes.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/14448#discussion_r108748825>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPLjCnVPUqa330cB1kFQwdh_Uhn25ks5rqp_RgaJpZM4MtVm0>
> .
>
